GameOfRunes/GameOfRunes/Scenes/GameScene.swift

```diff
@@ -9,28 +9,22 @@
 import SpriteKit
 import GameplayKit
 
-class GameScene: SKScene, GameSceneFacade {
-    private var gameEngine: GameEngineFacade!
+/**
+ `SKScene` for the single player game. The `SKScene` acts as the
+ root for all UI nodes.
+ - Manages state and scene transitions through `GameStateMachine`.
+ Any transitions related to states and scenes should be propagated to
+ this class (and in turn `GameStateMachine`) for handling.
+ - The immediate children of the `SKScene` should be `RootRenderNode`s,
+ which contain the logic used to implement the UI of the game.
+ - Game Loop originates in this class' `update()` method
+ */
+class GameScene: SKScene {
     private var lastUpdateTime: TimeInterval = 0.0
     private lazy var maximumUpdateDeltaTime: TimeInterval = { 1 / .init((view?.preferredFramesPerSecond ?? 60)) }()
-    private weak var gameStateMachine: GameStateMachine?
-    var center: CGPoint {
-        .init(x: frame.midX, y: frame.midY)
-    }
-
-    // layers
-    private var backgroundLayer: SKNode!
-    private var powerUpAnimationLayer: SKNode!
-    private var unitLayer: SKNode!
-    private var removalAnimationLayer: SKNode!
-    private var gestureLayer: SKNode!
-    private var playerAreaLayer: SKNode!
-    private var manaDropLayer: SKNode!
-    private var highestPriorityLayer: SKNode!
-    private(set) var playerAreaNode: PlayerAreaNode!
-    private(set) var playerEndPoint: SKSpriteNode!
-    private(set) var gestureAreaNode: GestureAreaNode!
-    private var bgmNode: SKAudioNode!
+    weak var gameStateMachine: GameStateMachine?
```

Comment:
@andyylam @JermyTan @brianyenna for the `GameStateMachine` do we want to have one for `SKScene` or one for each root node?

Reply:
Probably one for `SKScene`? I dun think the minimap needs `GameStateMachine`